### PR TITLE
Implementing zxcvbn instead of stupid rules

### PR DESCRIPTION
**Pull request by open-schnick**

### CODE REVIEW

**Comment by qvalentin** [comment]
@open-schnick you wanna push your commits?

**Comment by open-schnick** [comment]
> @open-schnick you wanna push your commits?

na

File: src/background/api/userInformation.ts [hunk 1]
```diff
@@ -11,7 +11,16 @@ export interface UserInformation {
...
     password?: string;
-    confirmationPassword?: string;
+    confirmationPassword?: string; // TODO remove this.
+}
+
+/**
+ * Interface describing the classic return value of the backend
+ */
+// FIXME does this type already exist?
+export interface UpdateUserErrorResponse {
```
Comment (qvalentin) [inline src/background/api/userInformation.ts]: Maybe the type should be generic for all backend responses.

File: src/components/pages/User/UserInformationInput.tsx [hunk 2]
```diff
@@ -160,101 +123,30 @@ export default function UserInformationInput(
                     placeholder="Must contain one number, uppercase & lowercase letter each"
```
Comment (qvalentin) [inline src/components/pages/User/UserInformationInput.tsx]: Placeholder should be updated

File: src/components/pages/User/UserInformationInput.tsx [hunk 3]
```diff
@@ -54,93 +34,76 @@ export default function UserInformationInput(
...
+        if (password) {
+            // check password strength
+            if (passwordStrength < REQUIRED_PASSWORD_STRENGTH) {
+                triggerAlert("Password is not strong enough");
+                return;
+            }
+
+            // check if passwords match
+            if (!passwordsMatch) {
+                triggerAlert("Password is not strong enough");
```
Comment (qvalentin) [inline src/components/pages/User/UserInformationInput.tsx]: Wrong Message: Passwords dont match

File: src/components/pages/User/Profile.tsx [hunk 4]
```diff
@@ -59,75 +56,81 @@ export default function Profile(): ReactElement {
...
-            newUser["password"] = await hashPassword(inputUser.password);
-            newUser["confirmationPassword"] = newUser["password"];
-        }
 
-        await changeUserInformation(newUser)
+        // trigger api call
+        await changeUserInformation(updatedUser)
             .then((res) => {
                 changeEditMode();
+                // FIXME this does never appear, because we rerender it and this gets lost
```
Comment (qvalentin) [inline src/components/pages/User/Profile.tsx:91]: Alerts should be defined global. Would need a lot more refacotring.

Comment (open-schnick) [reply src/components/pages/User/Profile.tsx:91]: Y. Dont know why we are not using something like this: https://www.npmjs.com/package/react-notifications

File: .husky/pre-push [hunk 5]
```diff
@@ -3,3 +3,4 @@
 
 eslint --max-warnings 0 src --ext .ts --ext .tsx --cache
 npm test  -- --watchAll=false
+git add -A src
```
Comment (qvalentin) [inline .husky/pre-push:6]: I dont think this is a good idea. Maybe just add the folder with the snapshots?

**Review by qvalentin — CHANGES_REQUESTED** [review]

File: .husky/pre-push (hunk 5 above)
Comment (open-schnick) [reply .husky/pre-push:6]: y i dont know it isnt that important

**Review by qvalentin — DISMISSED** [review]

**Review by qvalentin — APPROVED** [review]

**Pull request merged**